### PR TITLE
Add Optuna

### PR DESCRIPTION
**Pull request by sile**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by sile** [comment]
@g-votte @hvy @iwiwi @smly @toshihikoyanase agree to maintain this with me?

**Comment by sile** [comment]
@conda-forge/help-python this recipe is ready for your review.

File: recipes/optuna/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "7f7d92d8d1a3e0c5c3b92b4272a29b110a4a3d53e140a55e564c8756225a0c99"
+
+build:
+  number: 0
+  entry_points:
+    - optuna = optuna.cli:main
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (beckermr) [inline recipes/optuna/meta.yaml]: Can this be `noarch: python`?

**Comment by beckermr** [comment]
One question here. After that we can merge! Thanks for the contribution!

File: recipes/optuna/meta.yaml (hunk 1 above)
Comment (sile) [reply recipes/optuna/meta.yaml]: Yes, I added `noarch: python` (https://github.com/conda-forge/staged-recipes/pull/10081/commits/e55bb0a2035901450272991537092ec3b6313b9f).
Thanks!

File: recipes/optuna/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+build:
+  number: 0
+  entry_points:
+    - optuna = optuna.cli:main
+  noarch: python
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - alembic
```
Comment (chrisburr) [inline recipes/optuna/meta.yaml]: Are these all needed in host? Normally python and pip is enough

Comment (sile) [reply recipes/optuna/meta.yaml]: You're right. Thank you for pointing it out. I removed unnecessary packages (https://github.com/conda-forge/staged-recipes/pull/10081/commits/f705b93abfee2adfbaccc2c1b8406385c50d075d).

**Pull request merged**